ts/components/FiscalCodeComponent.tsx

```diff
@@ -221,42 +322,91 @@ const styles = StyleSheet.create({
 });
 
 export default class FiscalCodeComponent extends React.Component<Props> {
-  private renderFrontContent(profile: UserProfile, isLandscape: boolean) {
+  private renderItem(
+    content: string,
+    fullStyle: StyleProp<ViewStyle>,
+    landscapeStyle: StyleProp<ViewStyle>,
+    isLandscape: boolean
+  ) {
     return (
-      <React.Fragment>
-        <Text
-          bold={true}
-          style={[
+      <Text
+        bold={true}
+        robotomono={true}
+        style={[
+          isLandscape
+            ? [styles.landscapeText, landscapeStyle]
+            : [styles.fullText, fullStyle]
+        ]}
+      >
+        {content.toUpperCase()}
+      </Text>
+    );
+  }
+
+  private renderFrontContent(
+    profile: UserProfile,
+    municipality: MunicipalityState,
+    isLandscape: boolean
+  ) {
+    const fiscalCodeData = extractFiscalCodeData(
+      profile.fiscal_code,
+      municipality
+    );
+
+    return (
+      !pot.isLoading(municipality.data) && (
+        <React.Fragment>
+          {this.renderItem(
+            profile.fiscal_code,
+            styles.fullFiscalCodeText,
+            styles.landscapeFiscalCodeText,
             isLandscape
-              ? [styles.landscapeText, styles.landscapeFiscalCodeText]
-              : [styles.fullText, styles.fullFiscalCodeText]
-          ]}
-        >
-          {profile.fiscal_code.toUpperCase()}
-        </Text>
+          )}
 
-        <Text
-          bold={true}
-          style={[
+          {this.renderItem(
+            profile.family_name,
+            styles.fullLastNameText,
+            styles.landscapeLastNameText,
             isLandscape
-              ? [styles.landscapeText, styles.landscapeLastNameText]
-              : [styles.fullText, styles.fullLastNameText]
-          ]}
-        >
-          {profile.family_name.toUpperCase()}
-        </Text>
+          )}
 
-        <Text
-          bold={true}
-          style={[
+          {this.renderItem(
+            profile.name,
+            styles.fullNameText,
+            styles.landscapeNameText,
             isLandscape
-              ? [styles.landscapeText, styles.landscapeNameText]
-              : [styles.fullText, styles.fullNameText]
-          ]}
-        >
-          {profile.name.toUpperCase()}
-        </Text>
-      </React.Fragment>
+          )}
+
+          {pot.isSome(municipality.data) &&
```

Comment:
We could show CIT information even if **_municipality.data_** is not loaded yet.
In this way the card will be empty until municipality data is loaded. If we got an error or municipality data are not available we won't show nothing.
You could put `pot.isSome(municipality.data)` only for _place of birth_ and _provincia_ fields
What do you think?

Reply:
I personally preferred data being displayed all together but if loading time increase it could affect ux. I removed the condition reducing it to the city fields only within 6b423ea0dcb4cc5d946cd7170e15df2e186ac313.

Reply:
Yes I prefer data being display all at once too. In this specific case some data are retrieved from github and we can't predict if the resource is available and fast served